src/engine/server_iv.c

```diff
@@ -1176,6 +1186,7 @@ ds_iv_update(struct ds_iv_ns *ns, struct ds_iv_key *key, d_sg_list_t *value,
 	iv_sync.ivs_event = CRT_IV_SYNC_EVENT_UPDATE;
 	iv_sync.ivs_mode = sync_mode;
 	iv_sync.ivs_flags = sync_flags;
+	key->eph = crt_hlc_get();
```

Comment:
> It looks to me this assumes:
> 
> 1. IV updates are always initiated from same node; (synced HLC is guaranteed)

yes, ds_iv_update() is always being called by local engines to intiatate the update or sync.  Sorry, I actually not sure what you mean here. But this is the initiator HLC yes.


> 2. Out of order updates are disallowed not matter if the specific IV update is order sensitive or not. For example: for an integer IV, IV update is "+1" to current value, it actually doesn't care if the updates is in order or not.

Hmm, I actually do not see much of out of order case. Even this +1 case is a bit risky to me, especially if +1 or -1 are mixed. 

Anyway all our current cases needs to be in order.

> 
> I'm wondering should we enforce the assumptions in the generic IV mechanism? Or should we apply generation control for each IV individually?

I would rather to put it in order for now.  Though I can update it for each individually later if it is really needed.

Reply:
Let's take an example, I suppose the IV updates for "AGG_EPOCH_REPORT" are initiated from different nodes, and we shouldn't make them "in order", right? I assume the IV for reporting rebuild status are also updated from different nodes and not "in order"?

BTW, this patch is for pool IV only? Or it's a generic IV change for all kinds of IV? I didn't see how the 'eph' in IV entry is updated for container/rebuild/oid IV.

Reply:
> Let's take an example, I suppose the IV updates for "AGG_EPOCH_REPORT" are initiated from different nodes, and we shouldn't make them "in order", right? I assume the IV for reporting rebuild status are also updated from different nodes and not "in order"?

well, it is better to be in order, since these AGG_EPOCH are increasing anyway.  And also out of order is really very rare in these cases.

But blocking out of order update is very important for some cases like this one or container open/close or pool connect etc.   It might be ok for some cases indeed, but allowing it does not bring much benefits here IMHO,  So I would rather to "play safe" for now.

Btw: rebuild status can not be out of order, since there are final status report, and we can not let this one to be out of order with other update.